File: Spark/MinTempForStation/minTemp.scala
import org.apache.log4j.Level
import org.apache.log4j.Logger
import org.apache.spark.SparkContext

object min_temp extends App{

  def minimum(x:Float, y:Float) = {
    if (x > y) {y}
    else {x}
  }
  
 Logger.getLogger("org").setLevel(Level.ERROR)
 
 val sc = new SparkContext("local[*]","min_temp")
 val input = sc.textFile("/Users/gupta/Downloads/datasets/tempdata.csv")		// replace the path accordingly
 
 val mappedInput = input.map( x => (x.split(",")(0),
                                    x.split(",")(3).toFloat) )      // get the station_id and temp_recorded
                                    
 val minTemp = mappedInput.reduceByKey((x,y) => minimum(x,y))       
                                    
 val result = minTemp.collect()
 
  //displaying the result
 for (res <- result) {
   val station = res._1
   val mintemp = res._2
   println(s"$station minimum temperature: $mintemp F")
 }

}

File: Spark/MinTempForStation/minTemp.py
# find the minimum temperature for each station id

from pyspark import SparkContext

#Function to get the minimum
def minimum(x, y):
    if x > y:
        return y
    else:
        return x


sc = SparkContext("local[*]", "min_temp")
sc.setLogLevel("ERROR")
input_data = sc.textFile("/Users/gupta/Downloads/datasets/tempdata.csv")

mappedInput = input_data.map(lambda x: (x.split(",")[0],
                                        float(x.split(",")[3])))

minTemp = mappedInput.reduceByKey(lambda x, y: minimum(x, y))

result = minTemp.collect()

#displaying the results
for res in result:
    station = res[0]
    mintemp = res[1]
    print(f"{station} minimum temperature: {mintemp} F")
